tests/tests.rs

```diff
@@ -229,50 +229,71 @@ async fn test_on_disk_index() -> Result<()> {
     // We need separate processes because locks do not work within the same process
-    match unsafe { fork() } {
-        Ok(ForkResult::Parent { child }) => {
+    fork(
+        rusty_fork_test_name!(test_work_dir_lock),
+        rusty_fork_id!(),
+        |_| {},
+        move |c, _| {
+            create_dir_all(path.as_ref()).expect("failed to create path to init");
+            write(init_file, vec![]).expect("failed to create init file");
+            let runtime = Builder::new_current_thread()
+                .build()
+                .expect("failed to create runtime in parent");
             let now = Instant::now();
 
-            let storage_one = common::create_test_storage(&path, 1_000_000);
-            let res_one = storage_one.await;
+            let storage_one = common::create_test_storage(path.as_ref(), 1_000_000);
+            let res_one = runtime.block_on(storage_one);
             assert!(res_one.is_ok());
 
             let storage = res_one.unwrap();
-            let result = waitpid(child, None);
-            #[cfg(not(target_os = "macos"))]
-            assert_eq!(Ok(WaitStatus::Exited(child, 0)), result);
-            #[cfg(target_os = "macos")]
-            assert_eq!(
-                Ok(WaitStatus::Signaled(
-                    child,
-                    nix::sys::signal::Signal::SIGTRAP,
-                    false
-                )),
-                result
-            );
+            let exit = c
+                .wait_timeout(Duration::from_secs(MAX_PARENT_WAIT_SECONDS))
+                .expect("failed to wait for child");
+            if let Some(exit) = exit {
+                assert!(!exit.success());
+            } else {
+                c.kill().expect("failed to kill child process");
+            }
 
-            common::clean(storage, path)
-                .map(|res| res.expect("clean failed"))
-                .await;
+            runtime.block_on(
+                common::clean(storage, path.as_ref()).map(|res| res.expect("clean failed")),
+            );
 
             warn!("elapsed: {:.3}", now.elapsed().as_secs_f64());
-        }
-        Ok(ForkResult::Child) => {
-            sleep(Duration::from_secs(1)).await;
-            let storage_two = common::create_test_storage(&path, 1_000_000);
-            let _ = storage_two.await;
+        },
+        move || {
+            let runtime = Builder::new_current_thread()
+                .build()
+                .expect("failed to create runtime in child");
+            let mut attempt = 0;
+            while !read(&child_init_file).is_ok() {
+                assert!(attempt < MAX_CHILD_WAIT_COUNT);
+                attempt = attempt + 1;
+
+                std::thread::sleep(Duration::from_millis(200));
+            }
+            let storage_two = common::create_test_storage(child_path.as_ref(), 1_000_000);
+            let _ = runtime.block_on(storage_two);
             unreachable!("Second storage process must panic")
```

Comment:
Here is the problem: if `create_test_storage` not panicked, then the panic happens here. In that case parent process decided that the test passed, which is not correct.
I found this function in rust: https://doc.rust-lang.org/std/panic/fn.catch_unwind.html . It allows to catch the panic. So I suggest to wrap `runtime.block_on(storage_two)` with `catch_unwind` and return success if the code panicked and error when it is not